### PR TITLE
[Impeller] Fix stroke cap drawing not correct.

### PR DESCRIPTION
**Pull request by luckysmg**
Fix:
- https://github.com/flutter/flutter/issues/119810

### For `DisplayListTest.CanDrawCapsAndJoins`

**Before:**

**After:**

### For test: `DisplayListTest.CanDrawArc`

**Before:**

https://user-images.githubusercontent.com/49340347/217441198-563c7e12-900b-40a2-bdc9-1304950978ef.mp4

**After:**

https://user-images.githubusercontent.com/49340347/217441215-0febf1bf-7399-41c9-a1bb-0faa20d8c835.mp4

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt. See [testing the engine] for instructions on writing and running engine tests.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [x] All existing and new tests are passing.

### CODE REVIEW

File: impeller/entity/geometry.cc [hunk 1]
```diff
@@ -370,9 +371,15 @@ VertexBuffer StrokePathGeometry::CreateSolidStrokeVertices(
 
     // Generate start cap.
     if (!polyline.contours[contour_i].is_closed) {
-      auto cap_offset =
-          Vector2(contour.start_direction.y, -contour.start_direction.x) *
-          stroke_width * 0.5;
+      Vector2 direction;
+      if (cap == Cap::kButt) {
+        direction =
+            Vector2(contour.start_direction.y, -contour.start_direction.x);
```
[the remark this answers is not in the document]
Comment (luckysmg) [reply impeller/entity/geometry.cc:377]: When I junk set direction = `Vector2(-contour.start_direction.y, contour.start_direction.x)`

Render like this...Seems not correct

image

[the remark this answers is not in the document]
Comment (bdero) [reply impeller/entity/geometry.cc:377]: Ah, I just realized what's happening -- this is probably why I accidentally flipped it in the wrong direction when landing https://github.com/flutter/engine/pull/39124 in the first place...

This issue may occur for the other caps if the conditions are just right. I need to swing back around and rework the bridging solution, but this should be safe to land and cherry-pick. Thanks!

**Review by bdero — APPROVED** [review]
LGTM

**Comment by luckysmg** [comment]
Thx for reviewing this ^_^

**Comment by ZanderZhan** [comment]
@luckysmg Can this change be merged into the flutter stable branch as soon as possible?

**Comment by luckysmg** [comment]
@ZanderZhan See https://github.com/flutter/flutter/issues/124176.

**Pull request merged**